### PR TITLE
Better error reporting for `use`

### PR DESCRIPTION
**Pull request by edgarfgp**
## Description

Better error reporting for `use`

Continuation of https://github.com/dotnet/fsharp/pull/17779

### **Before**

Screenshot 2024-09-30 at 20 09 48

Screenshot 2024-09-30 at 20 10 12

### **After**

```fsharp
let x11 = 
    query { for c in [1..10] do
            use x = { new System.IDisposable with __.Dispose() = () }
            ^^^
            yield 1  }
    
    
let run r2 r3 =
    result {
        use b = r2
        ^^^
        return Ok 0
    }

```

## Checklist

- [x] Test cases added
- [x] Release notes entry updated

### CODE REVIEW

**Comment by edgarfgp** [comment]
This is ready

**Review by vzarytovskii — APPROVED** [review]

**Comment by edgarfgp** [comment]
Can I have a second review please ?

**Comment by edgarfgp** [comment]
Fixed merge conflicts and it is green again.

**Pull request merged**
